### ISSUE TITLE
Failed to restart CMake: connect ECONNREFUSED /tmp/projectname-cmake.sock

### ISSUE THREAD

**Issue by atar-axis** [go2sh/cmake-integration-vscode#1]
Hi there!
Thank you for your Extension, looks really great at a first glance.
Unfortunately I am not able to use it yet :disappointed: 

When I try to clean the project, the following message appears:
`Failed to restart CMake: connect ECONNREFUSED /tmp/ff1_selfcheck-3044-cmake.sock`

Furthermore hitting `F7` does nothing.

### Additional Context
```
flood@flood-r3 ~> cmake --version
cmake version 3.13.2

CMake suite maintained and supported by Kitware (kitware.com/cmake).
```
_CMakeLists.txt_
```
cmake_minimum_required( VERSION 3.12 )

# Define the compiler locations
set (CMAKE_C_COMPILER   gcc)
set (CMAKE_CXX_COMPILER g++)

# projectname
project (ff1_selfcheck)

# directories where to search for header files
include_directories(src)
include_directories(inc)

# cpp files to compile
set (srcs BrickChecker.cpp DeviceFileChecker.cpp ErrorDisplay_7seg.cpp ErrorDisplay.cpp ErrorDisplay_LEDButton.cpp main.cpp NetworkChecker.cpp)
list (TRANSFORM srcs PREPEND "src/") # cmake >= 3.12

# executable to create
add_executable (${PROJECT_NAME} ${srcs})

# link libraries
target_link_libraries(${PROJECT_NAME} ${CMAKE_SOURCE_DIR}/lib/${CMAKE_SYSTEM_PROCESSOR}/libbrick.a)
target_link_libraries(${PROJECT_NAME} ${CMAKE_SOURCE_DIR}/lib/${CMAKE_SYSTEM_PROCESSOR}/libaf3pihal.a)
target_link_libraries(${PROJECT_NAME} ${CMAKE_SOURCE_DIR}/lib/${CMAKE_SYSTEM_PROCESSOR}/libyaml-cpp.a)
target_link_libraries(${PROJECT_NAME} pthread)

add_definitions(-Wall -std=c++11 -g)
```
_settings.json_
```
{
        "files.associations": {
            "iostream": "cpp",
            "cstdint": "cpp",
            "ostream": "cpp",
            "vector": "cpp",
            "thread": "cpp",
            "cctype": "cpp",
            "clocale": "cpp",
            "cmath": "cpp",
            "cstdarg": "cpp",
            "cstddef": "cpp",
            "cstdio": "cpp",
            "cstdlib": "cpp",
            "cstring": "cpp",
            "ctime": "cpp",
            "cwchar": "cpp",
            "cwctype": "cpp",
            "array": "cpp",
            "atomic": "cpp",
            "*.tcc": "cpp",
            "chrono": "cpp",
            "list": "cpp",
            "unordered_map": "cpp",
            "exception": "cpp",
            "fstream": "cpp",
            "functional": "cpp",
            "initializer_list": "cpp",
            "iomanip": "cpp",
            "iosfwd": "cpp",
            "istream": "cpp",
            "limits": "cpp",
            "memory": "cpp",
            "numeric": "cpp",
            "optional": "cpp",
            "ratio": "cpp",
            "sstream": "cpp",
            "stdexcept": "cpp",
            "streambuf": "cpp",
            "string_view": "cpp",
            "system_error": "cpp",
            "type_traits": "cpp",
            "tuple": "cpp",
            "typeinfo": "cpp",
            "utility": "cpp",
            "algorithm": "cpp",
            "string": "cpp"
        },
        "C_Cpp.loggingLevel": "Debug",
        "C_Cpp.intelliSenseEngineFallback": "Enabled",
        "cmake.generator": "Unix Makefiles"
        }
```
_Log (Extension Host)_
```
[2019-01-13 17:07:01.576] [exthost] [info] extension host started
[2019-01-13 17:07:01.800] [exthost] [info] ExtensionService#_doActivateExtension vscode.debug-auto-launch {"startup":true,"activationEvent":"*"}
[2019-01-13 17:07:01.800] [exthost] [info] ExtensionService#loadCommonJSModule /usr/lib/code/extensions/debug-auto-launch/dist/extension
[2019-01-13 17:07:01.830] [exthost] [info] ExtensionService#_doActivateExtension vscode.emmet {"startup":true,"activationEvent":"*"}
[2019-01-13 17:07:01.830] [exthost] [info] ExtensionService#loadCommonJSModule /usr/lib/code/extensions/emmet/dist/extension
[2019-01-13 17:07:01.918] [exthost] [info] ExtensionService#_doActivateExtension vscode.git {"startup":true,"activationEvent":"*"}
[2019-01-13 17:07:01.918] [exthost] [info] ExtensionService#loadCommonJSModule /usr/lib/code/extensions/git/dist/main
[2019-01-13 17:07:02.276] [exthost] [info] ExtensionService#_doActivateExtension vscode.merge-conflict {"startup":true,"activationEvent":"*"}
[2019-01-13 17:07:02.276] [exthost] [info] ExtensionService#loadCommonJSModule /usr/lib/code/extensions/merge-conflict/dist/extension
[2019-01-13 17:07:02.296] [exthost] [info] ExtensionService#_doActivateExtension SanaAjani.taskrunnercode {"startup":true,"activationEvent":"*"}
[2019-01-13 17:07:02.296] [exthost] [info] ExtensionService#loadCommonJSModule /home/flood/.vscode-oss/extensions/sanaajani.taskrunnercode-0.3.0/out/extension
[2019-01-13 17:07:02.304] [exthost] [info] ExtensionService#_doActivateExtension vscode.configuration-editing {"startup":false,"activationEvent":"onLanguage:jsonc"}
[2019-01-13 17:07:02.304] [exthost] [info] ExtensionService#loadCommonJSModule /usr/lib/code/extensions/configuration-editing/dist/extension
[2019-01-13 17:07:02.331] [exthost] [info] ExtensionService#_doActivateExtension vscode.json-language-features {"startup":false,"activationEvent":"onLanguage:jsonc"}
[2019-01-13 17:07:02.331] [exthost] [info] ExtensionService#loadCommonJSModule /usr/lib/code/extensions/json-language-features/client/dist/jsonMain
[2019-01-13 17:07:02.545] [exthost] [info] ExtensionService#_doActivateExtension vscode.typescript-language-features {"startup":false,"activationEvent":"onLanguage:jsonc"}
[2019-01-13 17:07:02.545] [exthost] [info] ExtensionService#loadCommonJSModule /usr/lib/code/extensions/typescript-language-features/dist/extension
[2019-01-13 17:07:02.863] [exthost] [info] ExtensionService#_doActivateExtension ms-vscode.cpptools {"startup":true,"activationEvent":"workspaceContains:/.vscode/c_cpp_properties.json"}
[2019-01-13 17:07:02.863] [exthost] [info] ExtensionService#loadCommonJSModule /home/flood/.vscode-oss/extensions/ms-vscode.cpptools-0.20.1/out/src/main
[2019-01-13 17:07:04.398] [exthost] [info] ExtensionService#_doActivateExtension vscode.grunt {"startup":false,"activationEvent":"onCommand:workbench.action.tasks.runTask"}
[2019-01-13 17:07:04.398] [exthost] [info] ExtensionService#loadCommonJSModule /usr/lib/code/extensions/grunt/dist/main
[2019-01-13 17:07:04.419] [exthost] [info] ExtensionService#_doActivateExtension vscode.gulp {"startup":false,"activationEvent":"onCommand:workbench.action.tasks.runTask"}
[2019-01-13 17:07:04.419] [exthost] [info] ExtensionService#loadCommonJSModule /usr/lib/code/extensions/gulp/dist/main
[2019-01-13 17:07:04.481] [exthost] [info] ExtensionService#_doActivateExtension vscode.jake {"startup":false,"activationEvent":"onCommand:workbench.action.tasks.runTask"}
[2019-01-13 17:07:04.481] [exthost] [info] ExtensionService#loadCommonJSModule /usr/lib/code/extensions/jake/dist/main
[2019-01-13 17:07:04.524] [exthost] [info] ExtensionService#_doActivateExtension vscode.npm {"startup":false,"activationEvent":"onCommand:workbench.action.tasks.runTask"}
[2019-01-13 17:07:04.524] [exthost] [info] ExtensionService#loadCommonJSModule /usr/lib/code/extensions/npm/dist/main
[2019-01-13 17:07:05.329] [exthost] [info] ExtensionService#_doActivateExtension go2sh.cmake-integration-vscode {"startup":true,"activationEvent":"workspaceContains:**/CMakeLists.txt"}
[2019-01-13 17:07:05.329] [exthost] [info] ExtensionService#loadCommonJSModule /home/flood/.vscode-oss/extensions/go2sh.cmake-integration-vscode-0.1.3/out/src/extension
[2019-01-13 17:07:05.651] [exthost] [info] eager extensions activated
```
```
flood@flood-r3 ~> ps -aux | grep cmake
flood     1470  7.1  4.4 1243072 174048 ?      Sl   16:24   3:35 /usr/lib/chromium/chromium --ppapi-flash-version=32.0.0.101 --ppapi-flash-path=/usr/lib/PepperFlash/libpepflashplayer.so https://github.com/go2sh/cmake-integration-vscode
flood     4438  0.1  0.3  54536 13652 ?        S    17:07   0:00 cmake -E server --pipe=/tmp/ff1_selfcheck-4314-cmake.sock --experimental
flood     5057  0.0  0.0   8188   880 pts/0    S+   17:14   0:00 grep --color=auto cmake
```

**Comment by atar-axis** [comment]
The line after `workspaceManager -> restartClient -> yield client.start();` is never reached

**Comment by atar-axis** [comment]
this.pipe is btw `/test-9099-cmake.sock`, but...

```
flood@flood-r3:[~]$ cat /tmp/test-9099-cmake.sock 
cat: /tmp/test-9099-cmake.sock: No such device or address
```

**Comment by atar-axis** [comment]
Perfect! Thank you for your quick response and your hard work, I am at your disposal :)

**Comment by atar-axis** [comment]
It seems like you are a bit busy right now, so if I can help you in any way - let me know ;)

**Comment by go2sh** [comment]
Hi there,
I have been quite busy lately. Sometimes it's hard to find the right balance between work, family and spare time. :)
I just updated the extension. Can you try, if this version fixes your problem?

**Comment by atar-axis** [comment]
Hey there,
no problem :) thank you for your efforts!
`F7` works and I faced no crashes so far :+1: Thans!

**Unfortunately, I cannot switch between the projects now**

[peek 2019-01-22 15-38](https://user-images.githubusercontent.com/11148588/51542610-f30bba00-1e5b-11e9-80e1-2c140e5434a6.gif)

**Comment by go2sh** [comment]
:see_no_evil: Missed that, fixed that with 0.2.1.
Feel free to open more issues, when you find some more bugs. :-)
